sale_order_product_recommendation/wizards/sale_order_recommendation.py

```diff
@@ -63,11 +63,16 @@ def _recomendable_sale_order_lines_domain(self):
         sale_order_partner_field = (
             "partner_shipping_id" if self.use_delivery_address else "partner_id"
         )
-        other_sales = self.env["sale.order"].search(
-            [
-                (sale_order_partner_field, "child_of", partner.id),
-                ("date_order", ">=", start),
-            ]
+        # Search with sudo for get sale order from other commercials users
+        other_sales = (
+            self.env["sale.order"]
+            .sudo()
```

Comment:
If the partner is shared between companies, you could get sales from others that the current company, isn't it?